### DISCUSSION TITLE
Issue with --cdw=current in SSH

### DISCUSSION THREAD

**Discussion by gzagatti** [kovidgoyal/kitty#5892] [General]
In my local machine, my home folder is `/home/foo`. In directory `/home/foo/xy`, I am able to run the following command:
```sh
kitty @ launch --location=hsplit --cwd=current
```
which opens a new window in `/home/foo/xy`

I then log into a server 
```sh
kitty +kitten ssh myserver
```

In the server my username is `bar`, so my home folder is `/home/bar`.  In directory `/home/bar/ab`, I attempt the splitting command again:
```sh
kitty @ launch --location=hsplit --cwd=current
```

However, this raises an error:
```sh
cd: can't cd to /home/foo/xy
```

Now, let's say I start all over again. But this time, I am located in `/home` in my local machine. After I log into the server and move to directory `/home/bar/ab`, I issue the splitting command again:
```sh
kitty @ launch --location=hsplit --cwd=current
```
This time no error is raised, but the new split window ends up in the server in `/home`.

I would expect that with `--cwd=current`, the new window in the server would be opened in the directory that I am located in the server. Not the directory that I was in when I run the kitten `ssh`. Am I doing something wrong?

**Comment by kovidgoyal** [comment]
launch will open a new window in your local computer. if you want
automatic windows opened on the server, use the ssh kitten.

https://sw.kovidgoyal.net/kitty/kittens/ssh/

**Comment by gzagatti** [comment]
After a lot of experimenting, I have finally nailed the problem. 

My config file had `shell_integration no-rc no-cursor`. Previously I had problems with the rc part which led me to disable it. 

Now, after some testing I don't see anything breaking anymore so I changed to `shell_integration no-cursor`. The `launch` command works as expected. When invoked from the server, it will open the new window exactly in the same location as the server without raising any errors.

Thanks again for your help.
